main.py

```diff
@@ -101,11 +84,64 @@ def process_data():
     print(df4.to_string())
     df4.to_csv("filtered_data.csv")
 
+def cal_dividend_increament(div_obj, p):
+    # This function calculate the average exponential increments of the dividend
+    # and check wheter it is monotonically increasing and if it persistent with maximal time between dividends of 100 days
+    #  finding the events during the last p years
+    t_abs = div_obj.index.values
+    t_ns = np.datetime64('today')-t_abs
+    t_D = t_ns.astype('timedelta64[D]')
+    t = -t_D.astype('float64')/365
+    last_events = t>-p
+    t = t[last_events]
+    divs_values = div_obj.values[last_events]
+
+    # fiting to exponential model
+    divs_values_log = np.log(divs_values)
+    linreg = cp.stats.linregress(t, divs_values_log)
+    yearly_increment = np.exp(linreg.slope)-1
+
+    # checking if dividends are monotonically increasing
+    divs_shift = np.roll(divs_values, 1)
+    divs_shift[0] = divs_shift[1]
+    monotonic_test = divs_values - divs_shift >=0
+    is_monotonic = monotonic_test.all()
+
+    # checking if period between dividends exceeded 100 days
+    t_D = t*365
```

Comment:
because in python you can do anything you want in terms of coding style it was a mess for several years, then the pep8 standard was defined. IDEs such as pycharm enforce pep8 and issue a waning when you deviate from the standard.
https://peps.python.org/pep-0008/


for this example you should use: t_d_shift

"Function names should be lowercase, with words separated by underscores as necessary to improve readability.
Variable names follow the same convention as function names."